lib/Sema/CSDiagnostics.cpp

```diff
@@ -2710,6 +2710,31 @@ bool MissingArgumentsFailure::diagnoseTrailingClosure(ClosureExpr *closure) {
   return true;
 }
 
+bool DefaultArgumentTypeMismatch::diagnoseAsError() {
+  if (!isa<CallExpr>(getParentExpr()))
```

Comment:
Nit: All of the code until `emitDiagnostic` could be replaced with `getChoiceFor(getRawAnchor())`